src/Microsoft.AspNetCore.Routing/Dispatcher/TreeMatcher.cs

```diff
@@ -15,43 +14,41 @@
 using Microsoft.AspNetCore.Routing.Template;
 using Microsoft.AspNetCore.Routing.Tree;
 using Microsoft.Extensions.Internal;
-using Microsoft.Extensions.Logging;
 
 namespace Microsoft.AspNetCore.Routing.Dispatcher
 {
-    public class TreeDispatcher : DispatcherBase
+    public class TreeMatcher : MatcherBase
     {
         private bool _dataInitialized;
         private object _lock;
         private Cache _cache;
 
         private readonly Func<Cache> _initializer;
 
-        public TreeDispatcher()
+        public TreeMatcher()
         {
             _lock = new object();
             _initializer = CreateCache;
         }
 
-        public override async Task InvokeAsync(HttpContext httpContext)
+        public override async Task MatchAsync(MatcherContext context)
         {
-            if (httpContext == null)
+            if (context == null)
             {
-                throw new ArgumentNullException(nameof(httpContext));
+                throw new ArgumentNullException(nameof(context));
             }
 
-            EnsureServicesInitialized(httpContext);
+            EnsureServicesInitialized(context);
 
             var cache = LazyInitializer.EnsureInitialized(ref _cache, ref _dataInitialized, ref _lock, _initializer);
 
-            var feature = httpContext.Features.Get<IDispatcherFeature>();
-            var values = feature.Values?.AsRouteValueDictionary() ?? new RouteValueDictionary();
-            feature.Values = values;
+            var values = new RouteValueDictionary();
+            context.Values = values;
 
             for (var i = 0; i < cache.Trees.Length; i++)
             {
                 var tree = cache.Trees[i];
-                var tokenizer = new PathTokenizer(httpContext.Request.Path);
+                var tokenizer = new PathTokenizer(context.HttpContext.Request.Path);
 
                 var treenumerator = new Treenumerator(tree.Root, tokenizer);
```

Comment:
I feel like I should win an award for this. It's called `TreeEnumerator` in the old code. `Treenumerator` is so much obviously better.